src/tests.ts

```diff
@@ -0,0 +1,72 @@
+import sha256 from 'crypto-js/sha256.js';
+import logger from './utils/logger.js';
+import fs from "fs";
+import changesets from 'json-diff-ts';
+import fse  from 'fs-extra';
+
+export function checkASThashes(options: any, ASTType: string) {
+
+try {
+  let outputAST = fs.readFileSync('./zapps/'+options.inputFileName+'/'+ASTType+'/'+options.inputFileName+'_ast.json', 'utf8')
+  let outputASTJsonObject = JSON.parse(outputAST);
+  let outputASTHash = sha256(outputAST);
+  if (fs.existsSync('./truezapps/'+options.inputFileName+'/'+ASTType+'/'+options.inputFileName+'_ast.json')) 
+  {
+  let PreGeneratedAST = fs.readFileSync('./truezapps/'+options.inputFileName+'/'+ASTType+'/'+options.inputFileName+'_ast.json', 'utf8')
+  let PreGeneratedASTJsonObject = JSON.parse(PreGeneratedAST);
+  let PreGeneratedASTHash = sha256(PreGeneratedAST);
+  if(PreGeneratedASTHash.toString() !== outputASTHash.toString())
+  {
+    PreGeneratedASTHash = sha256(JSON.stringify(PreGeneratedASTJsonObject));
+    outputASTHash = sha256(JSON.stringify(outputASTJsonObject));
+    if(PreGeneratedASTHash.toString() !== outputASTHash.toString())
+    {
+      const VariyingData= changesets.diff(PreGeneratedASTJsonObject, outputASTJsonObject);
+      if(options.modifyAST == 'y')
+      {
+      changesets.applyChangeset(PreGeneratedASTJsonObject, VariyingData);
+      fs.writeFileSync('./truezapps/'+options.inputFileName+'/'+ASTType+'/'+options.inputFileName+'_ast.json', JSON.stringify(PreGeneratedASTJsonObject, null, 4), 'utf8');
+      logger.info('AST file mddified');
+      return 'false';    
+      }
+    }
+    else
+      return 'false';
```

Comment:
Should this be `true`? As I think we reach here if the hashes match after doing a `JSON.stringify`

Reply:
Right . We reach here when the ASTs match after trimming the extra whitespaces. Should we accept that as true?

Reply:
Yes I think so - the main information in the AST would remain the same so that counts as a pass imo